map/src/main/kotlin/edu/artic/map/MapActivity.kt

```diff
@@ -1,22 +1,121 @@
+import com.fuzz.rx.disposedBy
+import com.jakewharton.rxbinding2.view.visibility
 import edu.artic.base.utils.disableShiftMode
+import edu.artic.media.audio.AudioPlayerService
+import edu.artic.navigation.NavigationSelectListener
 import edu.artic.ui.BaseActivity
 import kotlinx.android.synthetic.main.activity_map.*
+import kotlinx.android.synthetic.main.minimal_exo_playback_control_view.view.*
 
+/**
+ * One of the four primary sections of the app. This Activity always hosts a [MapFragment].
+ *
+ * As a primary section, this class always binds and unbinds with the
+ * background [AudioPlayerService] in [onResume] and [onPause] (respectively
+ * speaking; see the linked audio player documentation for details).
+ */
 class MapActivity : BaseActivity() {
 
     override val layoutResId: Int
         get() = R.layout.activity_map
 
+    var boundService: AudioPlayerService? = null
+    private var audioIntent: Intent? = null
+
+    private val serviceConnection = object : ServiceConnection {
+        override fun onServiceDisconnected(name: ComponentName?) {
+            boundService = null
+        }
+
+        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+            val binder = service as AudioPlayerService.AudioPlayerServiceBinder
+            boundService = binder.getService()
+            setUpAudioServiceBindings()
+        }
+    }
+
+    private fun setUpAudioServiceBindings() {
+        boundService?.let { audioService ->
+            audioService.currentTrack.subscribe { audioFile ->
+                audioPlayer.trackTitle.text = audioFile.title
+            }.disposedBy(disposeBag)
+
+            audioService.audioPlayBackStatus
+                    .map { it is AudioPlayerService.PlayBackState.Playing }
+                    .bindToMain(audioPlayer.exo_pause.visibility())
+                    .disposedBy(disposeBag)
+
+            audioService.audioPlayBackStatus
+                    .map { it is AudioPlayerService.PlayBackState.Paused }
+                    .bindToMain(audioPlayer.exo_play.visibility())
+                    .disposedBy(disposeBag)
+
+            audioService.audioPlayBackStatus
+                    .map { it is AudioPlayerService.PlayBackState.Paused || it is AudioPlayerService.PlayBackState.Playing }
+                    .bindToMain(audioPlayer.visibility())
+                    .disposedBy(disposeBag)
+        }
+    }
+
+    private fun closeAudioPlayer() {
+        audioPlayer.animate()
+                .yBy(200f)
+                .setDuration(600)
+                .withEndAction {
+                    boundService?.stopPlayer()
+                    audioPlayer.visibility = View.INVISIBLE
+                    audioPlayer.y = audioPlayer.y - 200f
+                }
+                .start()
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        audioIntent = Intent(this, AudioPlayerService::class.java)
```

Comment:
if you are manually creating an intent of something outside the class its used for, it will lead to code maintenance issues. put this in a `companion object` of the `AudioPlayerService` that returns the intent. call it `newIntent()`